File: src/Appointment.java
import java.util.Date;

public class Appointment {
	private Date _date;
	private	int _employeeID;
	private int _customerID;
	private int _listingID;
	private Status _status;

	public enum Status{
		OPEN,
		CLOSED
	}

	Appointment(Date date, int employeeID, int customerID, int listingID ){
		_date = date;
		_employeeID = employeeID;
		_customerID = customerID;
		_listingID = listingID;
		_status = Status.OPEN;
	}

	/**
	* Cancel an appointment
	* @return true if appointment is canceled
	*/	
	public boolean cancelApointment(){
		_status = Status.CLOSED;
		return true;
	}	
	
}

File: server/models/appointment.js
class Appointment {
    constructor(listing_id, customer_id=-1, employee_id=-1, date) {
        this.listing_id = listing_id;
        this.customer_id = customer_id;
        this.employee_id = employee_id;
        this.date = date;
    }
}

function appointmentFromJson(json) {
    return new Appointment(json.listing_id, json.customer_id, json.employee_id, json.date);
}

exports.appointmentFromJson = appointmentFromJson;